ui/src/reducers/resource.js

```diff
@@ -4,140 +4,161 @@ import {
   APP_FORM_SETSTATUS,
   APP_FORM_RESET,
   APP_FIELD_SETVALUE
-} from "../actions/apps";
+} from '../actions/apps';
+import { DISMISS_ERROR, DISMISS_ALL_ERRORS } from '../actions/errors';
+import { wsError } from '../DataService';
 
 const defaultState = {
   isFetching: false,
   items: [],
-  fetchError: false,
+  errors: [],
   isCreating: false,
-  createError: false,
   isDeleting: false,
-  deleteError: false,
   isActioning: false,
-  actionError: false,
   isReading: false,
-  readingError: false,
   createClientAppDialog: {
     platforms: {},
     fields: {}
   }
 };
 
+const getErrors = (error, type, errors) => {
+  const index = errors.findIndex(e => e.type === type);
+  if (!error) {
+    if (index >= 0) {
+      return [...errors.slice(0, index), ...errors.slice(index + 1)];
+    }
+    return errors;
+  }
+  if (index >= 0) {
+    return [...errors.slice(0, index), { error, type }, ...errors.slice(index + 1)];
+  }
+  return [{ error, type }, ...errors];
+};
+
 const resourceReducer = actions => (state = defaultState, action) => {
   let index;
+  let errorsToDismiss;
+  let errors;
   switch (action.type) {
+    case DISMISS_ERROR:
+      if (action.errorMessage === wsError.message) {
+        delete wsError.message;
+      }
+      errorsToDismiss = state.errors.filter(e => e.error.message === action.errorMessage);
+      errors = [...state.errors];
+      errorsToDismiss.forEach(e => {
+        index = errors.findIndex(err => err === e);
+        errors = [...errors.slice(0, index), ...errors.slice(index + 1)];
+      });
+      return {
+        ...state,
+        errors
+      };
+    case DISMISS_ALL_ERRORS:
```

Comment:
is this actually used anywhere?

Reply:
it is called when route changes - https://github.com/aerogear/mobile-developer-console/pull/102/files#diff-0c044acc5817c9643bf3bb658b2d58c2R12